benchmarks/search/radius_search.cpp

```diff
@@ -7,56 +9,161 @@
 
 #include <chrono>
 
-static void
-BM_OrganizedNeighborSearch(benchmark::State& state, const std::string& file)
+void
+print_help()
 {
-  pcl::PointCloud<pcl::PointXYZ>::Ptr cloudIn(new pcl::PointCloud<pcl::PointXYZ>);
-  pcl::PCDReader reader;
-  reader.read(file, *cloudIn);
+  std::cout << "Usage: benchmark_radius_search <pcd filename>\n";
+  std::cout << "Additional argument options:\n";
+  std::cout << "Usage: benchmark_radius_search <pcd filename> <radius search> "
+               "<neighbor limit>\n";
+}
 
+static void
+BM_OrganizedNeighborSearch(benchmark::State& state,
+                           const pcl::PointCloud<pcl::PointXYZ>::Ptr cloudIn,
+                           const double searchRadius,
+                           const size_t neighborLimit)
+{
   pcl::search::OrganizedNeighbor<pcl::PointXYZ> organizedNeighborSearch;
   organizedNeighborSearch.setInputCloud(cloudIn);
 
-  double radiusSearchTime = 0;
-  std::vector<int> indices(cloudIn->size()); // Fixed indices from 0 to cloud size
-  std::iota(indices.begin(), indices.end(), 0);
   int radiusSearchIdx = 0;
 
+  pcl::Indices k_indices;
+  std::vector<float> k_sqr_distances;
   for (auto _ : state) {
-    int searchIdx = indices[radiusSearchIdx++ % indices.size()];
-    double searchRadius = 0.1; // or any fixed radius like 0.05
+    state.PauseTiming();
+    int searchIdx = radiusSearchIdx++ % cloudIn->size();
```

Comment:
```suggestion
    int searchIdx = radiusSearchIdx++ % cloudIn->size();
    while(!pcl::isFinite((*cloudIn)[searchIdx])) searchIdx = radiusSearchIdx++ % cloudIn->size();
```
The pcd files given to the executable contain some invalid points `(nan, nan, nan)`, so it might make sense to filter them out to get a more realistic benchmark. (same in the other benchmarks).